exercises/Stephen_Smith/ex01_basics/src/splitLineIntoWords.cpp

```diff
@@ -0,0 +1,49 @@
+#include "../include/splitLineIntoWords.h"
+
+
+
+// boolean expression to test if character is a space of hyphen
+bool spaceOrHyphen (char c)
+{
+    return (( c == '-') || (isspace(c)));
+}
+// boolean expression to ensure that character is neither a hyphen nor a space
+bool notSpaceOrHyphen ( char c)
+{
+    return !(( c == '-') || (isspace(c)));
+
+}
+// Splits a line into words using spaces and hyphens as delimiters between the words 
+std::vector<std::string> splitLineIntoWords(const std::string& line)
+{
+
+    // iterator over the string
+    std::string::const_iterator wordStart = line.begin();
```

Comment:
It's good to use `auto` as much as possible to avoid having to write these long-winded iterator types.